electron/js/menu/TrayIconHandler.js

```diff
@@ -0,0 +1,77 @@
+const {app, Menu, nativeImage, Tray} = require('electron');
+const config = require('./../config');
+const lifecycle = require('./../lifecycle');
+const locale = require('./../../locale/locale');
+const path = require('path');
+const windowManager = require('./../window-manager');
+
+function buildTrayMenu() {
+  const contextMenu = Menu.buildFromTemplate([
+    {
+      click: () => windowManager.showPrimaryWindow(),
+      label: locale.getText('trayOpen'),
+    },
+    {
+      click: async () => await lifecycle.quit(),
+      label: locale.getText('trayQuit'),
+    },
+  ]);
+
+  this.appIcon.on('click', () => windowManager.showPrimaryWindow());
+  this.appIcon.setContextMenu(contextMenu);
+  this.appIcon.setToolTip(config.NAME);
+}
+
+class TrayIconHandler {
+  constructor(platform, appIcon = new Tray(nativeImage.createEmpty())) {
+    this.appIcon = appIcon;
+    this.lastUnreadCount = 0;
+    this.platform = platform;
+
+    this.icons = {
+      badge: nativeImage.createFromPath(path.join(app.getAppPath(), 'img', 'taskbar.overlay.png')),
+      tray: nativeImage.createFromPath(path.join(app.getAppPath(), 'img', `tray.${this.defaultImageExtension}`)),
+      trayWithBadge: nativeImage.createFromPath(
+        path.join(app.getAppPath(), 'img', `tray.badge.${this.defaultImageExtension}`)
```

Comment:
Do we want to pull this out into a const for improved readability? Will be two instead of three lines as well.